src/window/index.ts

```diff
@@ -58,27 +63,58 @@ function createWindow(name: string, url: string, prefs: Electron.WebPreferences)
 
     return { action: 'deny' };
   });
+  window.webContents.on('did-fail-load', (event, errorCode, errorDescription, url) => {
+    console.log(`Failed to load ${ url }: ${ errorCode } (${ errorDescription })`);
+  });
   window.loadURL(url);
   windowMapping[name] = window.id;
+
+  return window;
 }
 
 /**
  * Open the preferences window; if it is already open, focus it.
  */
 export function openPreferences() {
-  let url = 'app://./index.html';
+  const webRoot = getWebRoot();
 
-  if (/^(?:dev|test)/i.test(process.env.NODE_ENV || '')) {
-    url = 'http://localhost:8888/';
-  }
-  createWindow('preferences', url, {
-    nodeIntegration:    true,
-    contextIsolation:   false,
-    enableRemoteModule: process.env?.NODE_ENV === 'test'
+  createWindow('preferences', `${ webRoot }index.html`, {
+    width:          940,
+    height:         600,
+    webPreferences: {
+      devTools:           !app.isPackaged,
+      nodeIntegration:    true,
+      contextIsolation:   false,
+      enableRemoteModule: process.env?.NODE_ENV === 'test'
+    },
   });
   app.dock?.show();
 }
 
+/**
+ * Open the first run window, and return once the user has accepted any
+ * configuration required.
+ */
+export async function openFirstRun() {
+  const webRoot = getWebRoot();
+  const window = createWindow('first-run', `${ webRoot }index.html#FirstRun`, {
```

Comment:
`src/pages/FirstRun.vue` is a _source_ file; it doesn't exist in the build.  Because we use [hash mode](https://github.com/rancher-sandbox/rancher-desktop/blob/e453e18e58ad5c02513670ce2d69ac50ad8fbff6/src/nuxt.config.js#L63) for the router (because we're not a real web server / browser), we navigate between pages by setting `document.location.hash`.

I'll add a comment though, because that's not obvious.

Reply:
Comment added in f717e7b870c177608a4fcb5a4daf3bc6f7ac8aed